packages/runtime/src/state/makeswift-api-client.ts

```diff
@@ -128,6 +197,43 @@ export type Dispatch = ThunkDispatch<State, unknown, Action>
 
 export type Store = ReduxStore<State, Action> & { dispatch: Dispatch }
 
-export function configureStore({ serializedState }: { serializedState?: SerializedState }): Store {
-  return createStore(reducer, APIResources.getInitialState(serializedState), applyMiddleware(thunk))
+// FIXME: this middleware can be removed once we've upgraded the builder to always provide
+// the locale when sending `SET_LOCALIZED_RESOURCE_ID` action
+function defaultLocaleMiddleware(
+  defaultLocale: string | undefined,
+): Middleware<Dispatch, State, Dispatch> {
+  return () => (next: ReduxDispatch<Action>) => {
+    return (action: Action): Action => {
+      switch (action.type) {
+        case ActionTypes.SET_LOCALIZED_RESOURCE_ID: {
+          return next(
+            action.payload.locale != null
+              ? action
+              : { ...action, payload: { ...action.payload, locale: defaultLocale } },
+          )
+        }
+      }
```

Comment:
Fill in `SET_LOCALIZED_RESOURCE_ID`'s `locale` argument w/ page locale if missing.